analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -72,9 +72,7 @@ public abstract class ParametersCorrectOrderBase<TArgumentSyntax> : SonarDiagnos
                     .Select(s => GetMethodDeclarationIdentifierLocation(s.GetSyntax()))
                     .WhereNotNull();
 
-                analysisContext.ReportIssue(Diagnostic.Create(SupportedDiagnostics[0], getLocationToReport(),
-                    additionalLocations: secondaryLocations,
-                    messageArgs: methodSymbol.Name));
+                analysisContext.ReportIssue(DiagnosticFactory.Create(SupportedDiagnostics[0], analysisContext.Compilation,  getLocationToReport(), secondaryLocations, methodSymbol.Name));
```

Comment:
`SupportedDiagnostics[0]` is an anti-pattern. Can you re-work this?

Reply:
I would not do this in the current PR. It's already big enough :)